site/src/components/Dashboard/useUpdateCheck.ts

```diff
@@ -0,0 +1,49 @@
+import { updateCheck } from "api/queries/updateCheck";
+import { useMemo, useState } from "react";
+import { useQuery } from "react-query";
+
+type UpdateCheckState = "show" | "hide";
+
+export const useUpdateCheck = (enabled: boolean) => {
+  const [dismissedVersion, setDismissedVersion] = useState(() =>
+    getDismissedVersionOnLocal(),
+  );
+  const updateCheckQuery = useQuery({
+    ...updateCheck(),
+    enabled,
+  });
+
+  const state: UpdateCheckState = useMemo(() => {
+    if (!updateCheckQuery.data) {
+      return "hide";
+    }
+
+    const isNotDismissed = dismissedVersion !== updateCheckQuery.data.version;
+    const isOutdated = !updateCheckQuery.data.current;
+    return isNotDismissed && isOutdated ? "show" : "hide";
+  }, [dismissedVersion, updateCheckQuery.data]);
+
+  const dismiss = () => {
+    if (!updateCheckQuery.data) {
+      throw new Error(
```

Comment:
Not sure here: is throwing an error safe? I can understand the error if the `dismiss` function gets called while the query has errored out, but this would also error out if the query response just happens to take a little longer than normal, right?

Reply:
There also isn't a loading state attached to the hook's return value, so anything using the hook wouldn't be able to tell when the dismiss is safe

Reply:
it is more a TS thing since you are not going to be able to dismiss it without the updateCheckQuery being completed but I can just do a `return;` 👍